### PR TITLE
fetch-in-react-practice - Brenda Trieu

### PR DESCRIPTION
**Pull request by brendatrieu**
https://user-images.githubusercontent.com/121825299/229930234-d6b1780b-4555-4770-a0f9-84c4804c6f6d.mov

### CODE REVIEW

File: fetch-in-react-practice/client/src/components/Todos.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        const data = await response.json();
+        setTodos(data);
+      } catch(err) {
+        handleError(err);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    fetchTodos();
+  }, [todos]);
```
Comment (RobertGardner) [inline fetch-in-react-practice/client/src/components/Todos.js:45]: Do you get a warning that you are missing `handleError` as a dependency? I wonder how smart the lint rule is...

File: fetch-in-react-practice/client/src/components/Todos.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        const response = await fetch(url('/api/todos'), {
+           method: "POST", headers: {"Content-Type": "application/json" }, body: JSON.stringify(newTodo)});
+        if (!response.ok) throw new Error('Status', response.status);
+        const data = await response.json();
+        setTodos(Array.prototype.concat(todos, data));
+      } catch(err) {
+        handleError(err);
+      }
+    }
+    fetchTodo();
```
Comment (RobertGardner) [inline fetch-in-react-practice/client/src/components/Todos.js:72]: There's no reason to do this embedded function thing on event handlers, only on `useEffect`. Event handlers can be async.

File: fetch-in-react-practice/client/src/components/Todos.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    const targetTodo = todos.filter((item) => item.todoId === todoId)[0];
+    const todoStatus = { isCompleted: !targetTodo.isCompleted };
+
+    async function fetchPatch() {
+      try{
+        const response = await fetch(url(`/api/todos/${todoId}`),
+        { method: "PATCH", headers: { "Content-Type": "application/json" }, body: JSON.stringify(todoStatus) });
+        if (!response.ok) throw new Error('Status', response.status);
+        const data = await response.json();
+        const newTodos = todos.map((todo) => (todo.todoId === todoId) ? (todo = targetTodo) : todo);
```
Comment (RobertGardner) [inline fetch-in-react-practice/client/src/components/Todos.js:107]: This is strange. Why are you reassigning to `todo`?
Oh, maybe you meant to just do `? targetTodo : todo`?

**Review by RobertGardner — CHANGES_REQUESTED** [review]
Looks very good, but I think you should fix the async function thing (in both places) before I approve.

File: fetch-in-react-practice/client/src/components/Todos.js (hunk 1 above)
Comment (brendatrieu) [reply fetch-in-react-practice/client/src/components/Todos.js:45]: No warning

**Review by RobertGardner — APPROVED** [review]

**Pull request merged**